### PR TITLE
refactor: remove global dynamic client

### PR DESCRIPTION
**Pull request by lianghao208**

### CODE REVIEW

File: pkg/clusterprovider/etcdcluster.go [hunk 1]
```diff
@@ -20,35 +20,44 @@ package clusterprovider
...
 	kstoneapiv1 "tkestack.io/kstone/pkg/apis/kstone/v1alpha1"
+	"tkestack.io/kstone/pkg/controllers/util"
 )
 
 // Cluster is an abstract, pluggable interface for etcd clusters.
 type Cluster interface {
+
 	// BeforeCreate does some things before creating the cluster
-	BeforeCreate() error
+	BeforeCreate(etcdCluster *kstoneapiv1.EtcdCluster) error
```
Comment (tangcong) [inline pkg/clusterprovider/etcdcluster.go:33]: etcdCluster -> cluster

File: pkg/clusterprovider/plugins.go [hunk 2]
```diff
@@ -27,38 +27,38 @@ import (
...
 )
 
-type EtcdFactory func(cluster *kstoneapiv1.EtcdCluster) (Cluster, error)
+type EtcdFactory func(cluster *ClusterContext) (Cluster, error)
 
 var (
-	mutex     sync.Mutex
-	providers = make(map[kstoneapiv1.EtcdClusterType]EtcdFactory)
+	mutex                sync.Mutex
+	EtcdClusterProviders = make(map[kstoneapiv1.EtcdClusterType]EtcdFactory)
```
Comment (tangcong) [inline pkg/clusterprovider/plugins.go:34]: Why change to uppercase？

**Review by tangcong — CHANGES_REQUESTED** [review]
good job. thanks.

**Review by tangcong — APPROVED** [review]

**Comment by engow** [comment]
LGTM👍  thxx
image

**Pull request merged**